common/metrics/scope.go

```diff
@@ -30,15 +31,26 @@ type metricsScope struct {
 	scope          tally.Scope
 	defs           map[int]metricDefinition
 	isDomainTagged bool
+	cachedScopes   map[string]Scope
```

Comment:
this also needs to be bounded and the bound prolly needs to be way smaller than the one in the metricsClient.